agent/api/statechange.go

```diff
@@ -349,16 +388,59 @@ func (change *TaskStateChange) String() string {
 	return res
 }
 
+// ToECSAgent converts the agent module level TaskStateChange to ecs-agent module level TaskStateChange.
+func (change *TaskStateChange) ToECSAgent() (*ecs.TaskStateChange, error) {
+	output := &ecs.TaskStateChange{
+		Attachment:         change.Attachment,
+		TaskARN:            change.TaskARN,
+		Status:             change.Status,
+		Reason:             change.Reason,
+		PullStartedAt:      change.PullStartedAt,
+		PullStoppedAt:      change.PullStoppedAt,
+		ExecutionStoppedAt: change.ExecutionStoppedAt,
+		MetadataGetter:     newTaskMetadataGetter(change.Task),
+	}
+
+	for _, managedAgentEvent := range change.ManagedAgents {
+		if mgspl := buildManagedAgentStateChangePayload(managedAgentEvent); mgspl != nil {
```

Comment:
nit: the `buildContainerStateChangePayload` signature returns an err, while the only failure case for the `buildManagedAgentStateChangePayload` seems to be when it shouldn't be reported to backend.  Are there any other error cases (`change.Container.Name` where container might be nil for example)?

Reply:
ACK, looking into this - may discuss more offline.

Reply:
Discussed offline: aligned that we will add a check that returns an error (which will be logged downstream) if the container state change lacks a container name.